bot/exts/moderation/infraction/_scheduler.py

```diff
@@ -257,6 +249,21 @@ async def apply_infraction(
                 else:
                     dm_result = f"{constants.Emojis.failmail} "
                     dm_log_text = "\nDM: **Failed**"
+                    if infr_type == "warning" and not ctx.channel.permissions_for(user).view_channel:
+                        failed = True
+                        log_title = "failed to apply"
+                        confirm_msg = (f":x: Failed to apply **warning** to {user.mention} "
+                                       "because DMing the user was unsuccessful")
```

Comment:
I'm open to changing this to something else like "Failed to apply warning to @mention because the user has their DMs disabled". 
I chose the current message because I felt it accounted for cases where users block the bot too, but I realize the distinction is irrelevant.